packages/rrweb-snapshot/src/snapshot.ts

```diff
@@ -1069,7 +1069,15 @@ export function serializeNodeWithId(
       stylesheetLoadTimeout,
       keepIframeSrcFn,
     };
-    for (const childN of Array.from(n.childNodes)) {
+    let childNodes = Array.from(n.childNodes) as Node[];
+    childNodes = childNodes.filter(
+      // filter out nodes that are assigned to a slot other places.
+      (childN) => !(childN as HTMLElement).assignedSlot,
+    );
+    if (n instanceof HTMLSlotElement)
+      // add nodes that are assigned to this slot and treat them as normal children.
+      childNodes = childNodes.concat(Array.from(n.assignedNodes()));
+    for (const childN of Array.from(childNodes)) {
```

Comment:
`instanceof HTMLSlotElement` doesn't work cross iframe, this should:
```suggestion
    if ('assignedNodes' in n)
      // add nodes that are assigned to this slot and treat them as normal children.
      childNodes = childNodes.concat(Array.from(n.assignedNodes()));
    for (const childN of Array.from(childNodes)) {
```